### PR TITLE
Allow specifying control plane service for actors and/or reminders

### PR DESCRIPTION
**Pull request by DeepanshuA**
# Description

This PR lets users configure their solution/ControlPlaneService for actors/reminders.
If nothing is given, traditional `placement` and inbuilt reminders will be used.

## Issue reference

Please reference the issue this PR links with: #7203 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@DeepanshuA see conflicts

**Comment by DeepanshuA** [comment]
> @DeepanshuA see conflicts

Resolved

File: pkg/actors/providers.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package actors
```
Comment (JoshVanL) [inline pkg/actors/providers.go:1]: Please add copyright headers.

File: pkg/actors/providers_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
+package actors
```
Comment (JoshVanL) [inline pkg/actors/providers_test.go:1]: Headers.

**Review by JoshVanL — CHANGES_REQUESTED** [review]

**Review by JoshVanL — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**